### PR TITLE
download PLAsTiCC data and convert to HDF5

### PR DESCRIPTION
**Pull request by helenqu**
# Adds PLAsTiCC data to Astropile

PLAsTiCC (Photometric LSST Astronomical Time-Series Classification Challenge) is a 2018 Kaggle challenge for astronomical time-series classification with a simulated spectroscopic dataset (~7k examples) for training and simulated WFD/DDF photometric dataset (~3.5M examples) for testing. The test set also includes 5 additional transient types (PISN, ILOT, CaRT, microlens-binary, microlens-string) not found in the training set.

## Links for reference:
- [Kaggle challenge](https://www.kaggle.com/c/PLAsTiCC-2018)
- [unblinded data (training + test)](https://zenodo.org/records/2539456)
- [modeling paper](https://arxiv.org/abs/1903.11756)

## HuggingFace Dataset Format
Each example contains:
- `object_id`: string
- `lightcurve`: object with
  - `band`: band of observation, arr(string) (one of `ugrizY`)
  - `flux`: flux value of observation, arr(float)
  - `flux_err`: flux uncertainty of observation, arr(float)
  - `time`: timestamp of observation in MJD, arr(float)
- `obj_type`: object type, string
- `redshift`: true redshift of the object, float
- `hostgal_photoz`: float
- `hostgal_specz`: float

### CODE REVIEW

**Comment by maja-jablonska** [comment]
Do we want to partition the lightcurves by heal-pixels as well? (for cross-matching purposes

**Comment by helenqu** [comment]
remaining TODOs:
- fix dataset loading according to Micah's new understanding of HF sequences
- separate the HDF5 files by healpix region
- baselines

**Comment by mb010** [comment]
Citation in [the plasticc.py](https://github.com/AstroPile/AstroPile_prototype/blob/add-plasticc/scripts/plasticc/plasticc.py) file is missing @helenqu .

File: scripts/plasticc/plasticc.py
Comment (jeraud) [inline scripts/plasticc/plasticc.py:88]: Are the bands just in integer values? And not strings (e.g., ugrizY)? Should we add a mapping or description somewhere of that?

**Review by jeraud — COMMENTED** [review]
Looks good to me at first sight! Added one additional comment/question, next to https://github.com/AstroPile/AstroPile_prototype/pull/22#issuecomment-2045084481.

File: scripts/plasticc/plasticc.py
Comment (mb010) [reply scripts/plasticc/plasticc.py:88]: @helenqu to fix this, I am storing the strings as an ordered sequence of strings matching the indicies of `bands`. :+1: See [this line](https://github.com/AstroPile/AstroPile_prototype/blob/90a1ef4e3c2dbc63c514b8bc8559e73666fc09b5/scripts/yse/yse.py#L167) (and related).

File: scripts/plasticc/plasticc.py
Comment (mb010) [inline scripts/plasticc/plasticc.py:82]: I think referring to `_BANDS` is more versatile than referring to `_PLASTICC_BANDS`. We can then try to make this a default.

**Review by mb010 — APPROVED** [review]
Looks good to me. :smiley: 
I'll leave it to @jeraud to approve. They might have more to say. I'll approve if they don't get back to it by Thursday. :+1:

**Comment by jeraud** [comment]
Looks good to me as well!

**Review by jeraud — APPROVED** [review]

**Pull request merged**
